lib/Lec6-BinaryHeap/binaryHeap.js

```diff
@@ -0,0 +1,41 @@
+let array_length;
+let heap_root = (input, i) => {
+    let left = 2 * i + 1;
+    let right = 2 * i + 2;
+    let maxValue = i;
+
+    if (left < array_length && input[left] > input[maxValue]) {
+        maxValue = left;
+    }
+    if (right < array_length && input[right] > input[maxValue]) {
+        maxValue = right;
+    }
+
+    if (maxValue != i) {
+        swap(input, i, maxValue);
+        heap_root(input, maxValue);
+    }
+}
+let swap = (input, index_One, index_Two) => {
+    var temp = input[index_One];
+
+    input[index_One] = input[index_Two];
+    input[index_Two] = temp;
+}
+
+let heapSort = (input) => {
+    array_length = input.length;
+    for (var i = Math.floor(array_length / 2); i >= 0; i -= 1) {
+        heap_root(input, i);
+    }
+    for (i = input.length - 1; i > 0; i--) {
+        swap(input, 0, i);
+        array_length--;
+
+        heap_root(input, 0);
+    }
+}
+let arr = [7, -2, 3, 12, 5, -1, 13, 7];
+heapSort(arr);
+console.log(arr);
+
```

Comment:
I can't find all methods of binary heap. All that you've described in `heapRoot()` method is the `sink()` method of binary heap. But it has also primary methods: `add()`, `get()`, `remove()` and helper method: `bubble()`.

- `add(el)`: should add `el` to the end of array and allow it to `bubble()` in dependence on conditions that are described in helper's `less(a, b)` function;
- `get()` : should return `first element` of queue (by other words - `first element` of your heap). How too do it:
 1. Swap `first element` with `last element`; 2. Apply `array.pop()`; 3. Allow `first element` to `sink()`
- `remove(el)`: should remove element from array if it exists and refactor the array in dependence on conditions of `less(a, b)` function. To do it: 1. Swap `element` with `last element`; 2. Apply `array.pop()`; 3. Allow swapped element to `bubble()` and `sink()`;
- `bubble(index)`: should compare element in position `index` with its parent and decide to `swap` them or not in dependence on result of `less(a, b)` function